src/aiida_quantumespresso/workflows/pw/relax.py

```diff
@@ -143,7 +143,21 @@ def get_builder_from_protocol(
             base.pw.parameters['CELL']['cell_dofree'] = 'shape'
 
         if relax_type in (RelaxType.CELL, RelaxType.POSITIONS_CELL):
-            base.pw.parameters['CELL']['cell_dofree'] = 'all'
+            x_dim, y_dim, z_dim = structure.pbc
+            if x_dim and not (y_dim or z_dim):
+                base.pw.parameters['CELL']['cell_dofree'] = 'x'
+            elif y_dim and not (x_dim or z_dim):
+                base.pw.parameters['CELL']['cell_dofree'] = 'y'
```

Comment:
Maybe you are more of an expert, but I'm wondering: assume you have a structure with `pbc == [False, True, False], and the following `CELL_PARAMETERS`:

```console
CELL_PARAMETERS angstrom
      13.866974650       0.0000000000       0.0000000000
      1.9334873250       3.3488982827       0.0000000000
      1.9334873250       1.1162994276       13.157371580
```
I’m wondering if I set cell_dofree to 'y', will only the second component of the second lattice parameter (i.e. 3.3488982827) be allowed to change? If only the second component is allowed to change, also the direction of the lattice vector will change, which could deform the chain structure unless all atoms are on the second lattice vector, no?

Reply:
In this scenario , and according to the manual `'y' : only the y component of axis 2 (v2_y) is moved` given the cell parameters you are giving there is an angle in that direction so perhaps your `pbe = True, False, True` is not what you want

Reply:
@mbercx  Perhaps if there are so many options for `cell_dofree ` the best for QEApp, is just to define the StructureData.pbc (via widget) and just use the logic from this PR externally (not in the RelaxWorkchain) to  override the QEworkchain. Since for the QEApp we might just need to consider only a few cases that are 1D (along one direction x,y,z) , 2D (in xy) and 3D. In anycase I know that `create_kpoints_from_distance` will take into account `StructureData.pbc` to generate the mesh. Let me know your opinion I can close the PR

Reply:
> In this scenario , and according to the manual `'y' : only the y component of axis 2 (v2_y) is moved` given the cell parameters` you are giving there is an angle in that direction so perhaps your `pbe = [True, False, True]` is not what you want

Yeah, it seems you typically would make sure the Xth lattice vector only has an X component. E.g. 

```
CELL_PARAMETERS angstrom
-24.711115283      0.0000            0.00000
0.00000           -24.808651431      0.00000
0.00000            0.000000          4.165100096
```

Would be a reasonable cell with `cell_dofree` = `'z'`. So a user has to be careful, but we can also help by adding validation on the `PwCalculation` class (but I won't subject you to that ^^).

> @mbercx Perhaps if there are so many options for cell_dofree  the best for QEApp, is just to define the StructureData.pbc (via widget) and just use the logic from this PR externally (not in the RelaxWorkchain) to override the QEworkchain. Since for the QEApp we might just need to consider only a few cases that are 1D (along one direction x,y,z) , 2D (in xy) and 3D. In anycase I know that create_kpoints_from_distance will take into account StructureData.pbc to generate the mesh. Let me know your opinion I can close the PR

I understand this will probably be quicker for your use case, and it possible avoids creating bugs in the main `aiida-quantumespresso` plugin that other users could suffer from. However, long term I think it's definitely something that should be implemented in the `aiida-quantumespresso` plugin, and since it relies on the `pbc` it doesn't affect any user running 3D structures at all. So I'd be inclined to push through and get it in here, with a rapid release promised by yours truly.

Reply:
@mbercx Ok, I will do the changes you suggested, I am pro removing the `parameters['SYSTEM']['assume_isolated'] = '2D'`

Reply:
@AndresOrtegaGuerrero if it's ok for you, I was already doing the changes and adding a test, but got distracted by https://github.com/aiidateam/aiida-core/issues/5967 when I actually tried to import a 2D structure. 😅

Reply:
@mbercx no problem! , thank you

Reply:
Hmm, seems I can't push to the `nanotech-empa` fork 😭. Could you give me "Write" permission? 😇 See

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/about-permissions-and-visibility-of-forks#about-permissions-of-forks

and

https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/repository-roles-for-an-organization#permissions-for-each-role